packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,184 @@
+import { Component, Element, h, Prop, Watch } from '@stencil/core';
+
+import {
+  PROGRESS_MODE,
+  PROGRESS_THICKNESS,
+  PROGRESS_TYPE,
+  TProgressMode,
+  TProgressThickness,
+  TProgressType,
+} from './bq-progress.types';
+import { validatePropValue } from '../../shared/utils';
+
+@Component({
+  tag: 'bq-progress',
+  styleUrl: './scss/bq-progress.scss',
+  shadow: true,
+})
+export class BqProgress {
+  // Own Properties
+  // ====================
+
+  private previousValue: number | null = null;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqProgressElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** It defines the mode of progress bar to display */
+  @Prop({ reflect: true }) mode: TProgressMode = 'determinate';
+
+  /** A number representing the current value of the progress bar */
+  @Prop({ reflect: true }) value = 0;
+
+  /** Progress bar thickness */
+  @Prop({ reflect: true }) thickness: TProgressThickness = 'medium';
+
+  /** Progress type */
+  @Prop({ reflect: true }) type: TProgressType = 'default';
+
+  /** If `true`, the progress bar will be displayed without border radius */
+  @Prop({ reflect: true }) borderShape: boolean = false;
```

Comment:
Perhaps it will be easy to understand if the `borderShape` accept two values: `square` | `rounded`. With default value to be `rounded` as per design.